File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/util/FileUtils.java
package de.jbamberger.offlineclient.util;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStreamReader;

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

public class FileUtils {

    public static String readFile(String path) throws IOException {
        return readFile(new File(path), "utf-8");
    }

    public static String readFile(File file, String encoding) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), encoding));
        String line;
        StringBuilder builder = new StringBuilder();
        while ((line = reader.readLine()) != null) {
            builder.append(line);
            builder.append("\n");
        }
        reader.close();
        return builder.toString();
    }
}

File: OfflineClient/cached-api/src/main/java/de/jbamberger/api/FileUtils.kt
package de.jbamberger.api

import java.io.*

/**
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

internal object FileUtils {

    @Throws(IOException::class)
    fun readFile(path: String): String {
        return readFile(File(path), "utf-8")
    }

    @Throws(IOException::class)
    fun readFile(file: File, encoding: String): String {
        val reader = BufferedReader(InputStreamReader(FileInputStream(file), encoding))
        var line = reader.readLine()
        val builder = StringBuilder()
        while (line != null) {
            builder.append(line)
            builder.append("\n")
            line = reader.readLine()
        }
        reader.close()
        return builder.toString()
    }
}
